adam/perception/high_level_semantics_situation_to_developmental_primitive_perception.py

```diff
@@ -278,6 +282,49 @@ def _get_relations_from_condition(
 
         return relations
 
+    def _find_perception_object_for_situation_object(
+        self, slot: SituationObject, entities_to_roles, action_roles_to_fillers
+    ) -> ObjectPerception:
+        if slot in entities_to_roles.keys():
+            filler_role_in_action = only(entities_to_roles[slot])
+            situation_object = only(action_roles_to_fillers[filler_role_in_action])
+            perception = self._objects_to_perceptions[situation_object]
+        else:
+            found_objects = []
+            for (
+                object_perception,
+                ontology_node,
+            ) in self._object_perceptions_to_ontology_nodes.items():
+                # check all the sub objects in the situation
+                if all(
+                    necessary_property in ontology_node.inheritable_properties
```

Comment:
check for properties using the methods on the `Ontology` class; they will handle inheritance for you.